File: _CodeTopics/LeetCode_other_problems/面试题/剑指Offer(第2版)/15/15.py
class Solution(object):
    def hammingWeight(self, n):
        """
        :type n: int
        :rtype: int
        """
        
        res = 0
        while n > 0:
            if n % 2 != 0:
                res += 1
            n /= 2
        return res
        
"""
https://leetcode-cn.com/submissions/detail/188947038/

601 / 601 个通过测试用例
状态：通过
执行用时: 12 ms
内存消耗: 13.1 MB

执行用时：12 ms, 在所有 Python 提交中击败了95.18%的用户
内存消耗：13.1 MB, 在所有 Python 提交中击败了22.68%的用户
"""

File: _CodeTopics/LeetCode_other_problems/面试题/剑指Offer(第2版)/15/tran/15.cpp
class Solution {
public:
    int hammingWeight(uint32_t n) {
        int res = 0;
        while (n > 0) {
            if (n % 2 != 0) {
                res += 1;
            }
            n /= 2;
        }
        return res;
    }
};

/*
https://leetcode-cn.com/submissions/detail/188947622/

601 / 601 个通过测试用例
状态：通过
执行用时: 4 ms
内存消耗: 5.9 MB

执行用时：4 ms, 在所有 C++ 提交中击败了39.40%的用户
内存消耗：5.9 MB, 在所有 C++ 提交中击败了52.96%的用户
*/
